### PR TITLE
Adds http_close_test.dart

### PR DESCRIPTION
**Pull request by zanderso**

### CODE REVIEW

File: test/http_close_test.dart [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+import "dart:async";
+import "dart:math";
+import "dart:typed_data";
+
+import "package:http_io/http_io.dart";
+import "package:test/test.dart";
+
+Future<Null> testClientAndServerCloseNoListen(int connections) {
+  Completer<Null> completer = new Completer();
```
Comment (rmacnak-google) [inline test/http_close_test.dart:13]: I think you need to say `new Completer<Null>()` for strong-mode.

**Review by rmacnak-google — APPROVED** [review]
Do tests in this repository also run when we make changes in dart-lang/sdk?

File: test/http_close_test.dart (hunk 1 above)
Comment (zanderso) [reply test/http_close_test.dart:13]: The analyzer is happy, so I'm going to leave this as-is. If there is a runtime strong-mode problem, I'll change all instances of this all in one go.

**Comment by zanderso** [comment]
These tests don't run on dart-lang/sdk commits. My plan is to DEPS this repo in to third_party/pkg_tested, in which case they'll run on sdk commits.

**Pull request merged**